instrumentation/net/http/otelhttp/internal/semconv/dup.go

```diff
@@ -0,0 +1,195 @@
+	attrs[1] = semconvNew.ServerAddress(host)
+	i := 2
+	if hostPort := requiredHTTPPort(req.TLS != nil, p); hostPort > 0 {
+		attrs[i] = semconvOld.NetHostPort(hostPort)
+		attrs[i+1] = semconvNew.ServerPort(hostPort)
+		i += 2
+	}
+	i += d.method(req.Method, attrs[i:])     // Max 3
+	i += d.scheme(req.TLS != nil, attrs[i:]) // Max 2
+
+	if peer, peerPort := splitHostPort(req.RemoteAddr); peer != "" {
+		// The Go HTTP server sets RemoteAddr to "IP:port", this will not be a
+		// file-path that would be interpreted with a sock family.
+		attrs[i] = semconvOld.NetSockPeerAddr(peer)
+		attrs[i+1] = semconvNew.NetworkPeerAddress(peer)
+		i += 2
+		if peerPort > 0 {
+			attrs[i] = semconvOld.NetSockPeerPort(peerPort)
+			attrs[i+1] = semconvNew.NetworkPeerPort(peerPort)
+			i += 2
+		}
+	}
+
+	if useragent := req.UserAgent(); useragent != "" {
+		// This is the same between v1.20, and v1.24
+		attrs[i] = semconvNew.UserAgentOriginal(useragent)
+		i++
+	}
+
+	if clientIP := serverClientIP(req.Header.Get("X-Forwarded-For")); clientIP != "" {
+		attrs[i] = semconvOld.HTTPClientIP(clientIP)
+		attrs[i+1] = semconvNew.ClientAddress(clientIP)
+		i += 2
+	}
+
+	if req.URL != nil && req.URL.Path != "" {
+		attrs[i] = semconvOld.HTTPTarget(req.URL.Path)
+		attrs[i+1] = semconvNew.URLPath(req.URL.Path)
+		i += 2
+	}
+
+	protoName, protoVersion := netProtocol(req.Proto)
+	if protoName != "" && protoName != "http" {
+		attrs[i] = semconvOld.NetProtocolName(protoName)
+		attrs[i+1] = semconvNew.NetworkProtocolName(protoName)
+		i += 2
+	}
+	if protoVersion != "" {
+		attrs[i] = semconvOld.NetProtocolVersion(protoVersion)
+		attrs[i+1] = semconvNew.NetworkProtocolVersion(protoVersion)
+		i += 2
+	}
+
+	return slices.Clip(attrs[:i])
+}
+
+func (d dupHTTPServer) method(method string, attrs []attribute.KeyValue) int {
+	if method == "" {
+		attrs[0] = semconvOld.HTTPMethod(http.MethodGet)
+		attrs[1] = semconvNew.HTTPRequestMethodGet
+		return 2
+	}
+	attrs[0] = semconvOld.HTTPMethod(method)
+	if attr, ok := methodLookup[method]; ok {
+		attrs[1] = attr
+		return 2
+	}
+
+	if attr, ok := methodLookup[strings.ToUpper(method)]; ok {
+		attrs[1] = attr
+	} else {
+		// If the Original method is not a standard HTTP method fallback to GET
+		attrs[1] = semconvNew.HTTPRequestMethodGet
+	}
+	attrs[2] = semconvNew.HTTPRequestMethodOriginal(method)
+	return 3
+}
+
+func (d dupHTTPServer) scheme(https bool, attrs []attribute.KeyValue) int { // nolint:revive
+	if https {
+		attrs[0] = semconvOld.HTTPSchemeHTTPS
+		attrs[1] = semconvNew.URLScheme("https")
+		return 2
+	}
+	attrs[0] = semconvOld.HTTPSchemeHTTP
+	attrs[1] = semconvNew.URLScheme("http")
+	return 2
+}
+
+// ResponseTraceAttrs returns trace attributes for telemetry from an HTTP response.
+//
+// If any of the fields in the ResponseTelemetry are not set the attribute will be omitted.
+func (d dupHTTPServer) ResponseTraceAttrs(resp ResponseTelemetry) []attribute.KeyValue {
+	attributes := []attribute.KeyValue{}
```

Comment:
This is going to rely on the compiler strategy to allocate space for the attributes. Likely over-allocating. Can we similar to above pre-allocate capacity? Or can we change above to also follow this?